### DISCUSSION TITLE
Icon not showing in TextBox

### DISCUSSION THREAD

**Discussion by mobynet1** [unoplatform/uno#18013] [Q&A] [answered]
Please forgive me for all of my newbie questions, but...

I looked inside of the Uno.Gallery to see how to display an icon inside of the text box and how it is done and I found this in the `TextBoxSamplePage` file:

```text
<TextBox PlaceholderText="With Icon"
			Style="{StaticResource FilledTextBoxStyle}"
			AutomationProperties.AutomationId="TextBox_Default_WithIcon_Material">
	<ut:ControlExtensions.Icon>
		<SymbolIcon Symbol="Find" />
	</ut:ControlExtensions.Icon>
</TextBox>
```

The Find icon is indeed rendered inside of the textbox as advertised.  When I do the same thing, the Icon does not render:

```text
<TextBox
	Background="{StaticResource MobyDarkToolBarBackground}"
	BorderBrush="Transparent"
	CornerRadius="4"
	FontFamily="{StaticResource PoppinsLightFontFamily}"
	FontSize="14"
	FontWeight="Thin"
	PlaceholderText="Search all folders"
	VerticalContentAlignment="Bottom"
	Width="377">
	<ut:ControlExtensions.Icon>
		<SymbolIcon
			Symbol="Find" />
	</ut:ControlExtensions.Icon>
</TextBox>
```

I have included the necessary namespace to my xaml file: `xmlns:ut="using:Uno.Themes` as shown in the sample page. I am presuming there is something else I need to include to get this to work.

Thanks in advance for your help!!

**Comment by morning4coffe-dev** [comment, marked as answer]
Hey @mobynet1,

I wasn't able to replicate the behavior you described with the code you provided. I am seeing the icon as expected:
![image](https://github.com/user-attachments/assets/94cd6351-c3b0-4357-939f-aaa4dac69184)
You are using the Material theme, right? Could you provide a more detailed reproduction of the issue or take a look at my [UnoApp2.zip](https://github.com/user-attachments/files/16740640/UnoApp2.zip) to see if you notice any differences? Changing the properties on the TextBox doesn't seem to affect the extension's behavior.

**Reply by mobynet1** [reply]
Greetings @morning4coffe-dev! Thanks for your response! I thought I had included `xmlns:um="using:Uno.Material"` in the preamble of the page where I was trying to do this, but I had placed it in another file. I am seeing the correct behavior now! Doh! What a difference one line of code can make!!! :-)

**Reply by morning4coffe-dev** [reply]
Any time! 😇 Glad we could find the issue together like this.

**Comment by mobynet1** [comment]
@morning4coffe-dev wins this week's prize! Thanks for your help!

**Comment by mobynet1** [comment]
I closed this too early!

I also see in the Material Design control document that the Material textbox also has a region under the control that allows for helper and error text messages. I looked at the ControlExtensions class and did not see how this can be handled. Where else do I need to look or is it not implemented?

**Reply by MartinZikmund** [reply]
This would be validation message area, but as validation APIs are not yet available in WinUI, we don't have support for them yet. You can alternatively add them under the control as part of your app layout, or can create a custom control/user control that includes such validation message area

**Comment by mobynet1** [comment]
Thank you @MartinZikmund
